gen/generic_json_gen

```diff
@@ -0,0 +1,328 @@
+}
+if (!exists($requiredAttributes->{"resource"}->{$A_R_UUID})) {
+	die("Attribute $A_R_UUID is required for this service");
+}
+if (!exists($requiredAttributes->{"user"}->{$A_U_UUID})) {
+	die("Attribute $A_U_UUID is required for this service");
+}
+
+# Get constant information for facility
+my $facility = perunServicesInit::getFacility;
+my $facilityName = $facility->getName;
+print("Generating JSON for facility '".$facilityName."'\n");
+
+my @destinations = $servicesAgent->getDestinations(service => $service->getId, facility => $facility->getId);
+my @facilityDestinations;
+for my $destination (@destinations) {
+	push(@facilityDestinations, $destination->getDestination);
+}
+
+# List all attributes for facility
+my %facility_attributes;
+for my $urn (keys %{$requiredAttributes->{facility}}) {
+	my $attr = $data->getFacilityAttributeValue(attrName => $urn);
+	$facility_attributes{$urn} = $attr;
+}
+
+
+# Init Structure
+my $output = {
+	"metadata"  => {
+		"version"  => $SCRIPT_VERSION,
+		"facility" => {
+			"facility_name" => $facilityName,
+			"destinations" => \@facilityDestinations,
+			"attributes"   => \%facility_attributes,
+		}
+	},
+	"users"     => {}, # Fill later
+	"vos"       => {}, # Fill later
+	"groups"    => {}, # Fill later
+	"resources" => {} # Fill later
+};
+
+
+# Generate structure with required attributes
+# For fast lookup of UUID <-> ID mapping
+my $lookup = {
+	"uuids" => {}, 		# UUID -> ID mapping
+	"ids"   => { 		# ID -> UUID mapping
+		"vo" 	=> {}, 	# VO ID -> UUID mapping
+		"group" => {}, 	# Group ID -> UUID mapping
+		"resource" => {}, # Resource ID -> UUID mapping
+	},
+};
+
+# VOs
+print("Generating JSON for VOs\n");
+my $voAgent = $agent->getVosAgent;
+for my $voId ($data->getVoIds()) {
+	my $vo = $voAgent->getVoById(id => $voId);
+	my $voUUID = $vo->getUuid;
+	if (!defined($voUUID)) {
+		$voUUID = $voId;
+	}
+	# Lookup update
+	$lookup->{uuids}->{$voUUID} = $voId;
+	$lookup->{ids}->{vo}->{$voId} = $voUUID;
+
+	# Load VO attributes
+	my %vo_attributes;
+	for my $urn (keys %{$requiredAttributes->{vo}}) {
+		my $value = $data->getVoAttributeValue(vo => $voId, attrName => $urn);
+		$vo_attributes{$urn} = $value;
+	}
+
+	my $voobj = {
+		"attributes" => \%vo_attributes,
+	};
+	$output->{vos}->{$voUUID} = $voobj;
+}
+
+
+# Groups
+print("Generating JSON for groups\n");
+my $groupAgent = $agent->getGroupsAgent;
+for my $gropuId ($data->getGroupIds()) {
+	my $group = $groupAgent->getGroupById(id => $gropuId);
+	my $groupUUID = $group->getUuid;
+	if (!defined($groupUUID)) {
+		$groupUUID = $gropuId;
+	}
+
+	# Lookup update
+	$lookup->{uuids}->{$groupUUID} = $gropuId;
+	$lookup->{ids}->{group}->{$gropuId} = $groupUUID;
+
+	# Load Group attributes (undef if not present)
+	my %group_attributes;
+	for my $urn (keys %{$requiredAttributes->{group}}) {
+		my $value = $data->getGroupAttributeValue(group => $gropuId, attrName => $urn);
+		$group_attributes{$urn} = $value;
+	}
```

Comment:
pretty sure we can use `$data->getAllGroupAttributes`, which should return all the required attributes I believe, for this (and all other cases), which should speed things up. Checkout `denbi_portal_compute_center` gen script